lib/movable_ink/aws/sns.rb

```diff
@@ -26,13 +39,53 @@ def notify_and_sleep(seconds, error_class)
 
       def notify_nsq_can_not_be_drained
         notify_slack(subject: 'NSQ not drained',
-                     message: 'Unable to drain NSQ')
+                     message: "Unable to drain NSQ for instance <https://#{my_region}.console.aws.amazon.com/ec2/v2/home?region=#{my_region}#Instances:search=#{instance_id};sort=instanceId|#{instance_id}>")
+        notify_pagerduty(region: my_region, instance_id: instance_id)
       end
 
-      def notify_slack(subject:, message:)
+      def notify_pagerduty(region:, instance_id:)
+        subject = "Unable to drain NSQ"
+        summary = "Unable to drain NSQ for instance #{instance_id} in region #{region}"
+
+        # the PagerDuty integration key is added to the payload in the AWS integration
+        json_message = {
+          pagerduty: {
+            event_action: 'trigger',
+            payload: {
+              source: 'MovableInkAWS',
+              summary: summary,
+              timestamp: Time.now.utc.iso8601,
+              severity: 'error',
+              component: 'nsq',
+              group: 'nsq',
+              custom_details: {
+                InstanceId: instance_id,
+              },
+            },
+            dedup_key: "nsq-not-draining-#{instance_id}",
+            links: [{
+              href: "https://#{region}.console.aws.amazon.com/ec2/v2/home?region=#{region}#Instances:search=#{instance_id};sort=instanceId",
+              text: 'View Instance'
+            }],
+          }
+        }.to_json
+
+        run_with_backoff do
+          subject = add_subject_info(subject: subject)
```

Comment:
Rather than making an assignment here, I think it's acceptable to pass `add_subject_info(subject: subject)` as the parameter in the `publish` call below.

Reply:
You got it